src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,81 @@
+package duke;
+
+import duke.task.*;
+import duke.exception.DukeException;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/**
+ * Encapsulates a storage object to handle file input/output.
+ */
+public class Storage {
+    private String filepath;
+
+    public Storage(String filepath) {
+        this.filepath = filepath;
+    }
+
+    public ArrayList<Task> load() throws DukeException {
+        ArrayList<Task> tasks = new ArrayList<>();
+
+        try {
+            File file = new File(filepath);
+            Scanner sc = new Scanner(file);
+
+            while (sc.hasNextLine()) {
+                Task task;
+                String instruction = sc.nextLine();
+                String[] lines = instruction.split(" \\| ");
+
+                switch (lines[0]) {
+                    case "T":
+                        task = new ToDo(lines[2]);
+                        break;
+                    case "D":
+                        task = new Deadline(lines[2], lines[3]);
+                        break;
+                    case "E":
+                        task = new Event(lines[2], lines[3]);
+                        break;
+                    default:
+                        throw new DukeException("The task does not exist.");
+                }
+
+                if (lines[1].equals("1")) {
+                    task.markAsDone();
+                }
+
+                tasks.add(task);
+            }
+
+            return tasks;
+
+        } catch (FileNotFoundException ex) {
+            throw new DukeException("File is not found");
```

Comment:
Hi, instead of throwing a new exception upon catching one, I would suggest catching both exceptions separately with multiple catch blocks and then specifying the course of action when the specific exception is encountered. 

Also, it is good practice to catch your exceptions where you expect it to be thrown, rather than making the method throws the exception and handing the responsibility to higher level code to catch it. This may sometimes make it difficult for other users to find the precise location of the error when testing your code. For your case, you caught the DukeException in the Duke constructor. Fortunately, there is only one statement in the "try" block so it is still not too difficult to identify the source of the thrown exception.